mllib/src/main/scala/org/apache/spark/mllib/optimization/LBFGS.scala

```diff
@@ -52,7 +52,8 @@ class LBFGS(private var gradient: Gradient, private var updater: Updater)
    * Restriction: numCorrections > 0
    */
   def setNumCorrections(corrections: Int): this.type = {
-    assert(corrections > 0)
+    require(corrections > 0, s"Number of corrections must be greater than 0," +
+      s" but got ${corrections}")
```

Comment:
Would the following style fit? If yes, this is slightly easier to read.

~~~scala
require(corrections > 0,
  s"Number of corrections must be greater than 0 but got ${corrections}")
~~~